buildkite/bazelci.py

```diff
@@ -1034,7 +1034,7 @@ def clone_git_repository(git_repository, platform, git_commit=None):
         execute_command(["git", "reset", remote_head, "--hard"])
     execute_command(["git", "submodule", "sync", "--recursive"])
     execute_command(["git", "submodule", "update", "--init", "--recursive", "--force"])
-    execute_command(["git", "submodule", "foreach", "--recursive", "git", "reset", "--hard"])
+    execute_command(["git", "submodule", "foreach", "--recursive", "--", "git", "reset", "--hard"])
     execute_command(["git", "clean", "-fdqx"])
     execute_command(["git", "submodule", "foreach", "--recursive", "git", "clean", "-fdqx"])
```

Comment:
Here too?

Reply:
And above in line 1022, too?

Reply:
I'm still in the process of testing the change locally, but I think this line is fine since it's identical to line 1022, which has been executed without problems.

Reply:
Actually, I think the code was doing this wrong and I don't see (from the docs) why the new version with the `--` would work - the man-page states that "command" is a string:

> foreach [--recursive] <command>
> Evaluates an arbitrary shell command in each checked out submodule.

So the correct code would be:

execute_command(["git", "submodule", "foreach", "--recursive", "git reset --hard"])

WDYT?

Reply:
Yeah, let's try it.